packages/swayze/lib/src/core/controller/selection/user_selections/fill_selection_state.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:meta/meta.dart';
+import 'package:swayze_math/swayze_math.dart';
+
+import 'model.dart';
+
+/// A immutable description of the disposition of the [FillSelectionModel], used
+/// for the drag and fill operations.
+///
+/// It allows at most one selection at a time, if any.
+///
+/// This is the portion of selections on [SwayzeSelectionController] that are
+/// created and changed by user gestures.
+///
+/// See also:
+/// - [SwayzeSelectionController] which keeps this state in a [ValueListenable].
+@immutable
+class FillSelectionState {
+  final FillSelectionModel? selection;
+
+  const FillSelectionState._({
+    required this.selection,
+  });
+
+  factory FillSelectionState.empty() => const FillSelectionState._(
+        selection: null,
+      );
```

Comment:
```suggestion
  const FillSelectionState.empty(): this(selection: null);
```

no need for a factory here

Reply:
Yeah, same thing in the end. But I'll remove the factory.